brave/src/main/java/brave/internal/codec/EntrySplitter.java

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright 2013-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.internal.codec;
+
+import brave.internal.Platform;
+
+/**
+ * Splits a character sequence that's in a delimited string, optionally trimming optional whitespace
+ * (<a href="https://httpwg.org/specs/rfc7230.html#rfc.section.3.2">OWS</a>) before or after
+ * delimiters.
+ *
+ * <p>This is intended to be initialized as a constant, as doing so per-request will add
+ * unnecessary overhead.
+ */
+public final class EntrySplitter {
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  public static final class Builder {
+    int maxEntries = Integer.MAX_VALUE;
+    char entrySeparator = ',', keyValueSeparator = '=';
+    boolean trimOWSAroundEntrySeparator = true, trimOWSAroundKeyValueSeparator = true;
+    boolean keyValueSeparatorRequired = true, shouldThrow = false;
```

Comment:
We are implementing the builder pattern here but those fields are accessible externally in the package. Doing this we could use the builder in the wrong way as this usage for instance `EntrySplitter.newBuilder().maxEntries = 0;`. Do we have any motivation for this? Otherwise, we could set as private.

Reply:
in Brave we try to be conscious of bytecode size as we cause bloat otherwise. That's why you'll see us almost never use the private modifier. We don't support other tools sharing the same package as we do, so it isn't a problem in practice.

Reply:
hmm, interesting. I will take a look at the readme that you sent. thanks for share. :)